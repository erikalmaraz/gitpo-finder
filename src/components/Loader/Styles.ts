import styled, { keyframes } from "styled-components";

export const LoaderWrapper = styled.section`
  display: flex;
  height: 100vh;
  width: 100%;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 0;
  background: ${({ theme }) => (theme === "light" ? "#fff" : "#1b1a1b")};
`;


const loadingAnimation = keyframes`
 0% { transform: scale(.5); }
 100% { transform: scale(.7); }
`;

export const LoaderImg = styled.img`
  width: 300px;
  animation-name: ${loadingAnimation};
  animation-duration: .8s;
  animation-iteration-count: infinite;
  filter: ${({ theme }) => (theme === "light" ? "none" : "invert(1)")};
 
`;
