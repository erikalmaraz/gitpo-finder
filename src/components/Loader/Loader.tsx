import React from "react";
import Loading from '../../resources/assets/images/loader.gif';
import * as S from "./Styles";
const Header = () => {
  return (
    <React.Fragment>
      <S.LoaderWrapper>
        <S.LoaderImg src={Loading} alt="Loader"/>
      </S.LoaderWrapper>
    </React.Fragment>
  );
};

export default Header;
