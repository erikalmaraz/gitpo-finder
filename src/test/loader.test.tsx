import React from "react";
import { render, screen } from "@testing-library/react";
import Loader from "../components/Loader/Loader";
import renderer from "react-test-renderer";

test("Should render Loader correctly", () => {
  const props = {
    theme: "light"
  }
  const tree = renderer.create(<Loader {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
