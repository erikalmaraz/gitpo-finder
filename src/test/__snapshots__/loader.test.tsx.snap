// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Loader correctly 1`] = `
<section
  className="sc-bdfBQB fguEFV"
>
  <img
    alt="Loader"
    className="sc-gsTEea hxkLSY"
    src="logo-01.png"
  />
</section>
`;
